ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -279,7 +279,7 @@ let check_vgpu ~__context ~op ~ref_str ~vgpus ~power_state =
   | `pool_migrate | `migrate_send
     when List.for_all is_migratable  vgpus
       && List.for_all is_suspendable vgpus -> None
-  | `suspend
+  | `suspend | `checkpoint
```

Comment:
The original condition is `| \`suspend`

Reply:
Sorry, I misunderstand OCaml match syntax.
> Pattern matching with guards corresponds to the evaluation of a conditional expression immediately after the pattern is matched. If this expression comes back true, then the expression associated with that pattern is evaluated, otherwise pattern matching continues with the following pattern.

So `| a | b when c` means the following evaluation steps:
1. match `a` or `b`
2. if matched, check condition `c`